server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1412,6 +1418,16 @@ private boolean checkForHypervisorIdUpdate(Consumer existing, ConsumerDTO incomi
         return false;
     }
 
+    private boolean isUsedByOwner(HypervisorIdDTO hypervisor, Owner owner) {
+        if (hypervisor == null) {
+            return false;
+        }
+        return consumerCurator.getHypervisorsBulk(
+            Collections.singletonList(hypervisor.getHypervisorId()), owner.getOwnerId())
+            .list()
+            .size() > 0;
```

Comment:
There already is a curator interface for querying with a single hypervisorId, without having to make a singleton list. The following should work:  `return this.consumerCurator.getHypervisor(hypervisor.getHypervisorId(), owner.getOwnerId()) != null`.

Reply:
Done.